File: core/src/oxygen/core/OCMain.java
/* (C) 2024 */
package oxygen.core;

import arc.*;
import arc.util.*;
import mindustry.core.UI;
import mindustry.game.*;
import mindustry.mod.*;
import oxygen.annotations.generator.*;
import oxygen.content.*;
import oxygen.ui.*;

/**
 * OCMain
 */
@ModMetaG(name = OCMain.name, minGameVersion = "146", author = OCMain.author,
    displayName = OCMain.displayName, version = OCMain.version, hidden = true)
public class OCMain extends Mod implements ModModifier {
  public static final String name = "oxygencore", author = "emmmm9o(novarc)", version = "1.0.0",
      displayName = "Oxygen Core";

  public OCMain() {
    OCVars.preinit();
    Events.on(EventType.FileTreeInitEvent.class, event -> {
      OCVars.init();
    });
  }

  @Override
  public void init() {
    Log.info("oxygen init");
    OCVars.laterInit();
  }

  @Override
  public void modifyUI(UI ui) {
    OCUI.load();
    OCUI.modifyUI(ui);
  }

  @Override
  public void loadContent() {
    OUnitTypes.load();
  }
}

File: core/src/main/kotlin/oxygen/core/OCMain.kt
package oxygen.core

import arc.util.*
import mindustry.*
import mindustry.mod.*
import oxygen.*
import oxygen.annotations.*
import oxygen.content.*
import oxygen.ui.dialogs.*

class OCMain : Mod() {
    companion object {
        const val NAME = "oxyc"
        const val DISPLAY_NAME = "Oxygen Core"

        @ModConfig(
            name = NAME,
            minGameVersion = "149",
            displayName = DISPLAY_NAME,
            author = "EmmmM9O",
            description = "Oxygen Core library",
            repo = "github.com/EmmmM9O/mindustry-oxygen-core",
            preloader = "oxygen.core.OCPreloader",
        )
        @JvmField
        var modConfig: ModConfig = ModConfig(NAME, "")
        var preloadFailed = false
        fun fail(reason: String) {
            val errorDialog = ErrorDialog(reason)
            Time.run(10f) {
                errorDialog.show()
            }
        }
    }

    init {
        if (Vars.mods.getMod(NAME) != null) {
            try {
                Class.forName("mindustry.mod.Preloader")
            } catch (e: ClassNotFoundException) {
                preloadFailed = true
                Log.err("mindustry.mod.Preloader not found.Please use Mindustry Oxygen")
                fail("@dialog.oxygen.error.preload")
            }
            if (!preloadFailed) Oxygen.init()
        }
    }

    override fun init() {
        if (preloadFailed) {
            return
        }
    }

    override fun loadContent() {
        if (preloadFailed) {
            return
        }
        OBlocks.init()
    }
}
